neqo-transport/src/send_stream.rs

```diff
@@ -340,7 +343,11 @@ impl TxBuffer {
         let new_retirable = self.ranges.acked_from_zero() - self.retired;
         let keep_len =
             self.buffered() - usize::try_from(new_retirable).expect("should fit in usize");
-        self.send_buf.truncate_front(keep_len);
+
+        // Truncate front
+        self.send_buf.rotate_left(self.buffered() - keep_len);
```

Comment:
Do we really need to rotate the buffer?  Can't we just provide the slice at the end when necessary and accept that sometimes that will result in a small frame?  It seems like if the frame assembly code was still hungry, it could ask this stream for a second slice.

This would be faster if you did:

```rust
  for i in 0..new_retirable {
    self.send_buf.pop_front();
   }
```
That's exactly how `truncate()` is implemented with `pop_back()`.

Reply:
That's how it's implemented?? How disappointing. Then yes your snippet is how we should do it. The rotation was just because I assumed a more efficient `truncate` implementation than that, not to avoid presenting small end slices to frame assembly (which yes will happen now).

I'll include this in a future "small-fixes" PR.